src/diffusers/pipelines/stable_cascade/pipeline_stable_cascade_prior.py

```diff
@@ -0,0 +1,629 @@
+            if prompt_embeds.shape != negative_prompt_embeds.shape:
+                raise ValueError(
+                    "`prompt_embeds` and `negative_prompt_embeds` must have the same shape when passed directly, but"
+                    f" got: `prompt_embeds` {prompt_embeds.shape} != `negative_prompt_embeds`"
+                    f" {negative_prompt_embeds.shape}."
+                )
+
+        if prompt_embeds_pooled is not None and negative_prompt_embeds_pooled is not None:
+            if prompt_embeds_pooled.shape != negative_prompt_embeds_pooled.shape:
+                raise ValueError(
+                    "`prompt_embeds_pooled` and `negative_prompt_embeds_pooled` must have the same shape when passed"
+                    f"directly, but got: `prompt_embeds_pooled` {prompt_embeds_pooled.shape} !="
+                    f"`negative_prompt_embeds_pooled` {negative_prompt_embeds_pooled.shape}."
+                )
+
+        if not isinstance(num_inference_steps, int):
+            raise TypeError(
+                f"'num_inference_steps' must be of type 'int', but got {type(num_inference_steps)}\
+                           In Case you want to provide explicit timesteps, please use the 'timesteps' argument."
+            )
+
+        if images:
+            for i, image in enumerate(images):
+                if not isinstance(image, torch.Tensor) and not isinstance(image, PIL.Image.Image):
+                    raise TypeError(
+                        f"'images' must contain images of type 'torch.Tensor' or 'PIL.Image.Image, but got"
+                        f"{type(image)} for image number {i}."
+                    )
+
+    @property
+    def guidance_scale(self):
+        return self._guidance_scale
+
+    @property
+    def do_classifier_free_guidance(self):
+        return self._guidance_scale > 1
+
+    @property
+    def num_timesteps(self):
+        return self._num_timesteps
+
+    def get_t_condioning(self, t, alphas_cumprod):
+        s = torch.tensor([0.003])
+        clamp_range = [0, 1]
+        min_var = torch.cos(s / (1 + s) * torch.pi * 0.5) ** 2
+        var = alphas_cumprod[t]
+        var = var.clamp(*clamp_range)
+        s, min_var = s.to(var.device), min_var.to(var.device)
+        ratio = (((var * min_var) ** 0.5).acos() / (torch.pi * 0.5)) * (1 + s) - s
+        return ratio
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        prompt: Optional[Union[str, List[str]]] = None,
+        images: Union[torch.Tensor, PIL.Image.Image, List[torch.Tensor], List[PIL.Image.Image]] = None,
+        height: int = 1024,
+        width: int = 1024,
+        num_inference_steps: int = 20,
+        timesteps: List[float] = None,
+        guidance_scale: float = 4.0,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        prompt_embeds: Optional[torch.FloatTensor] = None,
+        prompt_embeds_pooled: Optional[torch.FloatTensor] = None,
```

Comment:
```suggestion
        text_encoder_hidden_states: Optional[torch.FloatTensor] = None,
        prompt_embeds: Optional[torch.FloatTensor] = None,
```

I read your `encode_prompt` method and here is how `prompt_embeds` and `prompt_embeds_pools` are defined:

```python
prompt_embeds = text_encoder_output.hidden_states[-1]
prompt_embeds_pooled = text_encoder_output.text_embeds.unsqueeze(1)
```
basically, I think
1.  `prompt_embeds_pooled` here is the same as the `prompt_embeds` in most of the pipelines, (e.g. in stable diffusion https://github.com/huggingface/diffusers/blob/3cf4f9c7353b3d42c6b6b527df5c0359600bb197/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L404)  so let's rename it to `prompt_embeds` instead
2. `prompt_embeds` is a little bit special but is used in [unclip](https://github.com/huggingface/diffusers/blob/3cf4f9c7353b3d42c6b6b527df5c0359600bb197/src/diffusers/pipelines/unclip/pipeline_unclip.py#L159) and [Kandinsky](https://github.com/huggingface/diffusers/blob/3cf4f9c7353b3d42c6b6b527df5c0359600bb197/src/diffusers/pipelines/kandinsky/pipeline_kandinsky_prior.py#L333) so let's rename it to `text_encoder_hidden_states` to be consistent

Reply:
edit:
@sayakpaul point out these namings are consistent with SDXL so I think we don't need to rename in this PR